### PR TITLE
ComputeCappedWeakPlaneStress made more robust

### LINKED ISSUE #7960: Increase robustness of CappedWeakPlane plasticity
### Description of the enhancement or error report

(1) Check for precision loss every step during the Newton-Raphson procedure

(2) During constructor, check that tensile_strength + compressive_strength > smoothing tolerance
### Rationale for the enhancement or information for reproducing the error

(1) If the user sets their tolerance too low then the Newton-Raphson procedure will never converge because of precision loss.

(2) If the user sets their smoothing parameter too high then tensile and compressive failure will mix during the smoothing and potentially lead to no stresses being admissible
### Identified impact

(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)

more robust convergence

### PR DESCRIPTION
**Pull request by WilkAndy**
(1) A check for precision-loss is made during the return-map process.  If precision loss is detected, convergence is assumed.
(2) The constructor checks that tensile_strength + compressive_strength > smoothing_tol (previously the rhs of this inequality was zero).

Fixes #7960

### CODE REVIEW

**Comment by WilkAndy** [comment]
This PR failed due to unrelated fails in two tests using PETSc alt:

porous_flow:dirackernels.theis3.............................................................. FAILED (ERRMSG)
combined:contact_verification/patch_tests/cyl_3.frictionless_kin............................ FAILED (CSVDIFF)

@cpgr and/or I should investigate the theis3 test.

@acasagran wrote the cyl_3.frictionless_kin test, i think, so perhaps he/she would like to investigate?

**Comment by acasagran** [comment]
The cyl_3.frictionless_kin test is just diffing slightly. You can loosen the tolerances in the tests file to get it to pass. I would suggest changing abs_zero to 1e-4 and rel_err to 2e-5.

**Comment by WilkAndy** [comment]
OK, i will do that, thanks @acasagran

**Comment by WilkAndy** [comment]
I have modified the two offending tests.  We shall have to see if they pass now.

File: modules/porous_flow/tests/dirackernels/theis3.i [hunk 1]
```diff
@@ -171,8 +171,9 @@
   [./smp]
     type = SMP
     full = true
-    petsc_options_iname = '-ksp_type -pc_type -snes_atol -snes_rtol'
-    petsc_options_value = 'gmres asm 1e-8 1e-10'
+    petsc_options = '-snes_converged_reason -ksp_diagonal_scale -ksp_diagonal_scale_fix -ksp_gmres_modifiedgramschmidt -snes_linesearch_monitor'
+    petsc_options_iname = '-ksp_type -pc_type -sub_pc_type -sub_pc_factor_shift_type -pc_asm_overlap -snes_atol -snes_rtol -snes_max_it'
+    petsc_options_value = 'gmres      asm      lu           NONZERO                   2               1E-8       1E-10 20'
   [../]
```
Comment (cpgr) [inline modules/porous_flow/tests/dirackernels/theis3.i]: Is a 'hooley dooley' appropriate here?

**Comment by permcody** [comment]
Hooley Dooley is always appropriate!

File: modules/porous_flow/tests/dirackernels/theis3.i (hunk 1 above)
Comment (WilkAndy) [inline modules/porous_flow/tests/dirackernels/theis3.i]: Heheheh, this is a secret Andy recipe that always seem to work so far.

**Comment by WilkAndy** [comment]
@mangerij and @fdkong  , your new "periodic_boundary" test is failing with PETSc alt, and TBB .   I'm not sure whether we want to fix that in this PR, or create a new PR.  It's up to you what's easier for you.

**Comment by fdkong** [comment]
> @mangerij and @fdkong , your new "periodic_boundary" test is failing with PETSc alt, and TBB . I'm not sure whether we want to fix that in this PR, or create a new PR. It's up to you what's easier for you.

@WilkAndy, I will be making a simple example to replace the old one. I am sorry on this.

**Review by jwpeterson — CHANGES_REQUESTED** [review]
Please fix Python spacing, otherwise the changes look OK.

**Comment by WilkAndy** [comment]
Thanks @jwpeterson - python indentation is fixed.

**Review by jwpeterson — APPROVED** [review]

**Comment by WilkAndy** [comment]
Thanks everyone for your input here,

a

**Pull request merged**